pkg/provisioner/block/block.go

```diff
@@ -80,6 +87,70 @@ func roundUpSize(volumeSizeBytes int64, allocationUnitBytes int64) int64 {
 	return (volumeSizeBytes + allocationUnitBytes - 1) / allocationUnitBytes
 }
 
+func (block *blockProvisioner) waitForVolumeAvailable(volumeID *string, timeout time.Duration) error {
+	isVolumeReady := func() (bool, error) {
+		getVolumeRequest := core.GetVolumeRequest{VolumeId: volumeID}
+		ctx, cancel := context.WithTimeout(block.client.Context(), block.client.Timeout())
+		defer cancel()
+
+		getVolumeResponse, err := block.client.BlockStorage().GetVolume(ctx, getVolumeRequest)
+		if err != nil {
+			return false, err
+		}
+
+		if getVolumeResponse.LifecycleState == core.VolumeLifecycleStateAvailable {
+			return true, nil
+		}
+
+		if getVolumeResponse.LifecycleState == core.VolumeLifecycleStateFaulty {
+			return false, fmt.Errorf("Failed to provision volume. Volume: %v is in Faulty state", *volumeID)
+		}
+
+		if getVolumeResponse.LifecycleState == core.VolumeLifecycleStateTerminated {
+			return false, fmt.Errorf("Failed to provision volume. Volume: %v is in Terminated state", *volumeID)
+		}
+
+		if getVolumeResponse.LifecycleState == core.VolumeLifecycleStateTerminating {
+			return false, fmt.Errorf("Failed to provision volume. Volume: %v is in Terminating state", *volumeID)
+		}
+
+		return false, nil
+	}
+
+	ready, err := isVolumeReady()
```

Comment:
```go
return wait.PollImmediate(time.Second * 5, timeout, func() (bool, error) {
 	ready, err := isVolumeReady() 
	if err != nil {
		return false, fmt.Errorf("failed to provision volume %q: %v", *volumeID, err)
	}
	return ready, nil
})
```

`k8s.io/apimachinery/pkg/util/wait`

Reply:
I was halfway through writing this and then thought K8s would have something...